whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotFileGenerator.java

```diff
@@ -36,88 +39,66 @@ public class SnapshotFileGenerator {
     private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotFileGenerator.class);
     private static final int QUEUE_CAPACITY = 1000;
 
+    private final WhoisObjectRepository whoisObjectRepository;
+
     private final Dummifier dummifierNrtm;
     private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
     private final RpslObjectEnqueuer rpslObjectEnqueuer;
     private final SnapshotFileSerializer snapshotFileSerializer;
-    private final SnapshotFileRepository snapshotFileRepository;
-    private final SnapshotGenerationWindow snapshotGenerationWindow;
     private final SourceRepository sourceRepository;
+    private final NrtmFileRepository nrtmFileRepository;
+    private final DateTimeProvider dateTimeProvider;
+
 
     public SnapshotFileGenerator(
         final Dummifier dummifierNrtm,
         final NrtmVersionInfoRepository nrtmVersionInfoRepository,
         final RpslObjectEnqueuer rpslObjectEnqueuer,
+        final WhoisObjectRepository whoisObjectRepository,
         final SnapshotFileRepository snapshotFileRepository,
+        final NrtmFileRepository nrtmFileRepository,
+        final DateTimeProvider dateTimeProvider,
         final SnapshotFileSerializer snapshotFileSerializer,
-        final SnapshotGenerationWindow snapshotGenerationWindow,
         final SourceRepository sourceRepository
     ) {
         this.dummifierNrtm = dummifierNrtm;
         this.nrtmVersionInfoRepository = nrtmVersionInfoRepository;
         this.rpslObjectEnqueuer = rpslObjectEnqueuer;
         this.snapshotFileSerializer = snapshotFileSerializer;
-        this.snapshotFileRepository = snapshotFileRepository;
-        this.snapshotGenerationWindow = snapshotGenerationWindow;
         this.sourceRepository = sourceRepository;
+        this.whoisObjectRepository = whoisObjectRepository;
+        this.nrtmFileRepository = nrtmFileRepository;
+        this.dateTimeProvider = dateTimeProvider;
     }
 
-    public void createSnapshots(final SnapshotState state) {
-        // Get last version from database.
+    public void createSnapshot() {
+        final Stopwatch stopwatch = Stopwatch.createStarted();
+        final List<NrtmSource> sources = getSources();
```

Comment:
It will happen only in initialization creation of source, May be I can update it to createAndGetResources But still I prefer getSources as ultimately it is getting sources. But if you want I can rename